### PR TITLE
Identify/fix some mcard inlines

### PR DESCRIPTION
**Pull request by cadmic**
I started matching more mcard functions and I think these are correct now, although I'm not sure on `mcardFinishFile` vs `mcardFinishGame`. Highlights:

* `UnkInlinemCardReplaceFileBlock` -> `mcardReadAnywhereNoTime`
* `UnkInlinemCardReplaceFileBlock2`, `UnkInlinemCardVerifyChecksumFileHeader2` -> `mcardWriteAnywhereNoTime`
* `UnkInlinemCardVerifyChecksumFileHeader`, `mcardGameEraseInline` -> `mcardReadyFile`
* `UnkINlinemCardVerifyChecksumFileHeader3`, `UnkINlinemCardVerifyChecksumFileHeader4` -> `mcardFinishFile`
* `mcardGetFileTime`: missing `pMCard->saveToggle == true` check
* `mcardFileRelease`: should call `mcardGameRelease`

### CODE REVIEW

File: src/emulator/mcardGCN.c [hunk 1]
```diff
@@ -287,12 +257,12 @@ static bool mcardReplaceFileBlock(MemCard* pMCard, s32 index) {
 
     simulatorPrepareMessage(S_M_CARD_SV09);
 
-    if (!UnkInlinemCardReplaceFileBlock2(pMCard, index << 13)) {
+    if (!mcardWriteAnywhereNoTime(pMCard, index << 13, 0x2000, pMCard->writeBuffer)) {
```
Comment (Yanis42) [inline src/emulator/mcardGCN.c:260]:
```suggestion
    if (!mcardWriteAnywhereNoTime(pMCard, index << 13, CARD_SYSTEM_BLOCK_SIZE, pMCard->writeBuffer)) {
```

File: src/emulator/mcardGCN.c [hunk 2]
```diff
@@ -263,11 +233,11 @@ static bool mcardReplaceFileBlock(MemCard* pMCard, s32 index) {
     s32 checksum1;
     s32 checksum2;
     char buffer[0x2000];
-    s32 pad;
+    s32 pad[2];
 
     memcpy(buffer, pMCard->writeBuffer, 0x2000);
 
-    if (!UnkInlinemCardReplaceFileBlock(pMCard)) {
+    if (!mcardReadAnywhereNoTime(pMCard, 0x4000, 0x2000, pMCard->writeBuffer)) {
```
Comment (Yanis42) [inline src/emulator/mcardGCN.c:240]:
```suggestion
    if (!mcardReadAnywhereNoTime(pMCard, 0x4000, CARD_SYSTEM_BLOCK_SIZE, pMCard->writeBuffer)) {
```
maybe that 0x4000 is ``CARD_SYSTEM_BLOCK_SIZE * 2``? not sure if that makes sense though

File: src/emulator/mcardGCN.c [hunk 3]
```diff
@@ -263,11 +233,11 @@ static bool mcardReplaceFileBlock(MemCard* pMCard, s32 index) {
     s32 checksum1;
     s32 checksum2;
     char buffer[0x2000];
```
Comment (Yanis42) [inline src/emulator/mcardGCN.c:235]:
```suggestion
    char buffer[CARD_SYSTEM_BLOCK_SIZE];
```

File: src/emulator/mcardGCN.c [hunk 4]
```diff
@@ -189,9 +195,10 @@ static bool mcardCalculateChecksumFileBlock2(MemCard* pMCard, s32* checksum) {
...
 
-static bool mcardSaveChecksumFileHeader(MemCard* pMCard, char* buffer, u32 unused, u32 unused2) {
-    char buffer2[8192];
+static bool mcardSaveChecksumFileHeader(MemCard* pMCard, char* buffer) {
+    char buffer2[0x2000];
     s32 checksum;
+    s32 pad[2];
 
     memcpy(buffer2, pMCard->writeBuffer, 0x2000);
     memcpy(pMCard->writeBuffer, buffer, 0x2000);
```
Comment (Yanis42) [inline src/emulator/mcardGCN.c:204]:
```suggestion
    char buffer2[CARD_SYSTEM_BLOCK_SIZE];
    s32 checksum;
    s32 pad[2];

    memcpy(buffer2, pMCard->writeBuffer, CARD_SYSTEM_BLOCK_SIZE);
    memcpy(pMCard->writeBuffer, buffer, CARD_SYSTEM_BLOCK_SIZE);
```

**Review by Yanis42 — APPROVED** [review]
``dolphin/card.h`` got some macros that could be used here, I made suggestions but idk if that makes sense or not

**Comment by cadmic** [comment]
I definitely agree on `CARD_SYSTEM_BLOCK_SIZE` (though maybe we want a shorter name) but there are quite a lot of places it should be used, much more than the ones you pointed out. Is it already if we leave it for a cleanup PR?

**Comment by cadmic** [comment]
Yeah sory, I think I'd like to do a cleanup pass after the rest of mcard is decompiled, because it's a bit hard to tell how things are used right now

**Pull request merged**